src/Npgsql/BackendMessages/ErrorOrNoticeMessage.cs

```diff
@@ -7,6 +7,7 @@ namespace Npgsql.BackendMessages
     class ErrorOrNoticeMessage
     {
         internal string Severity { get; }
+        internal string? InvariantSeverity { get; }
```

Comment:
Why is it nullable?

Reply:
Because it is only provided starting from PostgreSQL 9.6.

I'd be fine with making it non-nullable and saying that it's the user's responsibility to be aware of this - this makes the API better going forward for sure, and is aligned with how nullability is being annotated in various other places. Let me know what you think.

Reply:
> I'd be fine with making it non-nullable and saying that it's the user's responsibility to be aware of this

I think that sounds reasonable. 

Though, given that EOL for 9.6 is still two years away and PostgreSQL-like databases on earlier versions of the protocol would also be (very mildly) inconvenienced: 

Could we just shim for older versions by null-coalescing to `Severity`? E.g.:

```c#
return new ErrorOrNoticeMessage(
    severity, invariantSeverity ?? severity, code, message,
```

Reply:
Great solution, am going to do this.